pkg/util/sequence/sequence.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sequence
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/builtins"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+)
+
+// GetSequenceFromFunc extracts a sequence name from a FuncExpr if the function
+// takes a sequence name as an arg. Returns the name of the sequence and a bool
+// representing whether a sequence name was found or not.
+func GetSequenceFromFunc(funcExpr *tree.FuncExpr) (*string, error) {
+	searchPath := sessiondata.SearchPath{}
+
+	def, err := funcExpr.Func.Resolve(searchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	fnProps, overloads := builtins.GetBuiltinProperties(def.Name)
+	if fnProps != nil && fnProps.TrackSequenceDeps {
+		found := false
+		for _, overload := range overloads {
+			// Find the overload that matches funcExpr.
+			if funcExpr.ResolvedOverload().Types.Match(overload.Types.Types()) {
```

Comment:
Is there a chance that there might be ambiguity in which overload we pick?

Reply:
It shouldn't be possible, it compares the types of the arguments, it doesn't make sense to have two overloads with the exact same arguments.

Reply:
Good point